docs/cookbook/using-docker.md

```diff
@@ -0,0 +1,343 @@
+
+Next, we are specifying that the database should communicate over port `27017`. This is the port typically used by Apostrophe but might need to be altered to match your environment.
+
+Finally, we add a volume for the MongoDB storage engine to write files into. You shouldn't need to change this.
+
+Looking at the `web:` container, we aren't passing an image but instead passing `build`. Within this, we are adding `context: .` which specifies we should build the image for this container from the `Dockerfile` in the same directory.
+
+The next two lines, starting with `ports:`, list the ports that the container should listen through, in this case, the typical port 3000.
+
+The `environment:` key lists environment variables that will get passed into the container. We could set the value of these here but are using a `.env` file instead.
+
+The `depends_on:` key indicates that our Apostrophe instance requires the presence of the `db` container that we created first.
+
+Finally, much like with the database, we are persisting a volume for any uploads to be written into. Without this, any uploads would be lost the next time we deployed.
+
+### Creating the `.env` file
+The last file we need to create before bringing our project up is a `.env` file at the root of our project containing the environmental variables. In this example file, we are assuming that you are hosting on a single server. Therefore we are setting the `APOS_CLUSTER_PROCESSES` environment variable to `2` to ensure that there is availability in case of a restart due to a crash. This number could be increased depending on your server.
+
+<AposCodeBlock>
+
+```bash
+NODE_ENV=production
+APOS_MONGODB_URI=mongodb://db:27017/apostrophe
+APOS_CLUSTER_PROCESSES=2
+```
+
+<template v-slot:caption>
+  .env
+</template>
+
+</AposCodeBlock>
+
+The only other line that might need alteration is the `APOS_MONGODB_URI` if your database needs to listen on a different port.
+
+### Spinning our project up
+Bringing our project up in Docker is a two-step process. First, from the CLI run `docker compose build`. This will create our project image. You should see commands from your `Dockerfile`, messages from the npm install, and then the familiar messages from Apostrophe as it builds the assets.
+
+When this finishes, you can run the command `docker compose up`. This will bring your project up and if you are using the defaults, allow you to access the site at `http://localhost:3000`. At this point, you won't be able to log in because it is a fresh database. To do this, you need to open a session with your container running Apostrophe. With both containers still running, give the following command from your terminal.
+
+```bash
+docker container ls
+```
+
+This will list all of your containers. Pick the one running your project image, not the database container and get the container id. Use it to run this command:
+
+```bash
+docker exec -it <container ID> /bin/sh
```

Comment:
Does docker compose not offer a more convenient path to doing this without looking up IDs?

Reply:
Hmm - I could alter the compose file to add a `container-name`.